spec/lib/whitehall_importer/integrity_checker_spec.rb

```diff
@@ -92,6 +92,45 @@
       integrity_check = WhitehallImporter::IntegrityChecker.new(edition)
       expect(integrity_check.valid?).to be true
     end
+
+    it "returns true even if there is a mismatch in inline atttachment URL filesize" do
+      body_field = double(:body_field,
+                          payload: {
+                            title: "Edition with attachment",
+                            base_path: "/edition-with-attachment",
+                            details: {
+                              body: "<p>Some other text</p><p><span class=\"gem-c-attachment-link\"><a class=\"govuk-link\" href=\"filename.pdf\" target=\"_blank\">Test File</a> (<span class=\"gem-c-attachment-link__attribute\"><abbr title=\"Portable Document Format\" class=\"gem-c-attachment-link__abbr\">PDF</abbr></span>, <span class=\"gem-c-attachment-link__attribute\">391 KB</span>, <span class=\"gem-c-attachment-link__attribute\">9 pages</span>)</span></p>",
+                              },
+})
+      document_type = build(:document_type, contents: [body_field])
+      edition_with_inline_attachment = build(:edition,
+                                             document_type: document_type,
+                                             tags: {
+                                               primary_publishing_organisation: [SecureRandom.uuid],
+                                               organisations: [SecureRandom.uuid],
+                                             })
+
+      publishing_api_item_with_inline_attachment = {
+        content_id: edition_with_inline_attachment.content_id,
+        base_path: "/edition-with-attachment",
+        title: "Edition with attachment",
+        description: edition_with_inline_attachment.summary,
+        document_type: edition_with_inline_attachment.document_type.id,
+        schema_name: edition_with_inline_attachment.document_type.publishing_metadata.schema_name,
+        details: {
+          body: "<p>Some other text</p><p><span class=\"attachment-inline\"><a href=\"/filename.pdf\">Test File</a> (<span class=\"type\">PDF</span>, <span class=\"file-size\">391KB</span>, <span class=\"page-length\">9 pages</span>)</span></p>",
+        },
+        links: {
+          primary_publishing_organisation: edition_with_inline_attachment.tags["primary_publishing_organisation"].to_a,
+          organisations: edition_with_inline_attachment.tags["organisations"].to_a + edition_with_inline_attachment.tags["primary_publishing_organisation"].to_a,
+        },
+      }
+
+      stub_publishing_api_has_item(publishing_api_item_with_inline_attachment)
+
+      integrity_check = WhitehallImporter::IntegrityChecker.new(edition_with_inline_attachment)
+      expect(integrity_check.valid?).to be true
```

Comment:
This test seems pretty heavy in the amount of things going on. I think it's a sign that there are perhaps too many things going on in this class. Maybe it's an idea to move the body checking part of the integrity checker to a nested class that we can test individually with body text input? That might make it easier to cover testing things like dealing with different HTML or spacing which I don't think we really test against.

Reply:
I agree, testing this proved to be the most complex part of this PR.  I've nested a new class, and simplified the tests.